admin-dev/themes/new-theme/js/pages/tax-rules-group/TaxRulesGroupForm.js

```diff
@@ -0,0 +1,346 @@
+        }
+      });
+    } else {
+      if ($(taxRulesGroupFormMap.stateFormRow).is(':visible')) {
+        this._resolveStateSelectVisibility([]);
+      }
+    }
+  }
+
+  /**
+   * Handles edit tax rule button click to add form action and tax rule values
+   *
+   * @param event
+   *
+   * @private
+   */
+  _handleEditTaxRuleClick(event) {
+    event.preventDefault();
+
+    const editLink = $(event.target).closest('a').attr('href');
+
+    if (editLink === this.taxRulesFormAction) {
+      this._hideTaxRulesForm();
+
+      return;
+    }
+
+    this._showTaxRulesForm(editLink, true);
+  }
+
+  /**
+   * Handles add new tax rule button click to show form with predefined field values
+   *
+   * @param event
+   *
+   * @private
+   */
+  _handleAddTaxRuleClick(event) {
+    event.preventDefault();
+    const addLink = $(taxRulesGroupFormMap.addLink).attr('href');
+
+    if (addLink === this.taxRulesFormAction) {
+      this._hideTaxRulesForm();
+
+      return;
+    }
+
+    this._showTaxRulesForm(addLink);
+  }
+
+  /**
+   * Hides tax rule form block
+   *
+   * @private
+   */
+  _hideTaxRulesForm() {
+    const $taxRuleForm = $(taxRulesGroupFormMap.taxRuleForm);
+
+    if ($taxRuleForm.is(':visible')) {
+      $taxRuleForm.fadeOut();
+    }
+
+    $taxRuleForm.attr('action', '');
+    this.taxRulesFormAction = '';
+  }
+
+  /**
+   * Shows tax rule form and fills it with ajax response or empty data
+   *
+   * @param link
+   * @param loadEditData
+   * @param loadEmptyData
+   *
+   * @private
+   */
+  _showTaxRulesForm(link, loadEditData = false, loadEmptyData = true) {
+    const $taxRuleForm = $(taxRulesGroupFormMap.taxRuleForm);
+    const $taxRuleFormHiddenContent = $(taxRulesGroupFormMap.taxRulesHiddenContent);
+    $taxRuleFormHiddenContent.hide();
+
+    this._enableSpinner();
+
+    if ($taxRuleForm.hasClass('hidden-element')) {
+      $taxRuleForm.removeClass('hidden-element');
+    }
+
+    $taxRuleForm.fadeIn();
+
+    this.taxRulesFormAction = link;
+    $taxRuleForm.attr('action', link);
+
+    if (loadEditData) {
+      $.ajax({
+        url: link,
+        method: 'GET',
+        dataType: 'json',
+      }).then(response => {
+        this._setTaxRuleInformation(response);
+        this._disableSpinner();
+        $taxRuleFormHiddenContent.show();
+      }).catch((response) => {
+        if (typeof response.responseJSON !== 'undefined') {
+          showErrorMessage(response.responseJSON.message);
+          this._disableSpinner();
+          this._hideTaxRulesForm();
+        }
+      });
+
+      return;
+    }
+
+    if (loadEmptyData) {
+      this._setTaxRuleInformation(null);
+    }
+
+    this._disableSpinner();
+  }
+
+  /**
+   * Resolves visibility of state select and fills it with ajax response data if visible
+   *
+   * @param response
+   * @param fadeSpeed
+   *
+   * @private
+   */
+  _resolveStateSelectVisibility(response, fadeSpeed = 400) {
```

Comment:
Do you expect `response` here? It seems that you are expecting states instead. Please name your variables with specific names, that gives some context about it.